draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,453 @@
+---
+Start date: 2024-05-24
+Author(s): [simeng-li]
+Revision: 1
+---
+
+# Sign-in experience and experience APIs
+
+## 1. Abstract
+
+Sign-in experience (SIE) is the process of user authentication in the Logto platform. It includes the user interface and experience APIs that allow end-users to interact with the Logto platform for identity verification and profile management.
+
+## 2. Motivation
+
+In Logto, we support a variety of modern authentication methods such as password, social login, and multi-factor authentication (MFA). Developers may require a flexible and customizable sign-in experience to meet their specific use cases.
+
+Current version of `Experience APIs` in Logto are neither well-documented nor publicly available for the developers. This document aims to provide a comprehensive overview of the SIE model in Logto and introduce a new set of `Experience APIs` that allow developers to build a customized sign-in experience for their users.
+
+### 3.1 Terminology
+
+- **Sign-in experience(SIE)**: A user's experience when interacting with Logto platform. Including the front-end user interface and the back-end API calls.
+- **Experience APIs**: A set of APIs that allow end-users to interact with the Logto for identity verification and profile management.
+- **Interaction session**: A session that is initiated when a user starts interacting with the Logto platform. The session will be terminated once the user logs out or the session expires.
+- **Experience app**: The front-end application that provides the user interface for the user interaction flow.
+- **Social/Single Sign-On(SSO) identity**: A third-party identity provider that allows users to sign in to the Logto platform using their social media or enterprise SSO accounts.
```

Comment:
1. Better to split into two items.
2. The definition is confusing. I assume that "identity" here is some data Logto stores, not an identity provider.

```suggestion
- **Social identity:** The data used to identify a user within a social identity provider, such as Google.
- **SSO (Single Sign-On) identity:** The data used to identify a user within an identity provider that is configured as an enterprise SSO connector in Logto.
```